lib/consumer.ex

```diff
@@ -276,26 +288,42 @@ defmodule GenRMQ.Consumer do
 
   @doc false
   @impl GenServer
-  def handle_info(:init, state) do
-    state =
-      state
-      |> get_connection()
-      |> open_channels()
-      |> setup_consumer()
+  def handle_info(
+        {:DOWN, ref, :process, _pid, reason},
+        %{module: module, config: config, running_tasks: running_tasks} = state
+      ) do
+    if Map.has_key?(running_tasks, ref) do
```

Comment:
What happens in this case to the message? Will it be rejected/nacked?

Reply:
Good call....we would probably want some sort of configuration so the user can decide what happens to the message if the task fails. Perhaps we also provide the option to send the message to the deadletter exchange. Also on that same note, we may also want some configuration around how long the Tasks take to complete. Currently it is set to the `async_nolink` value of 5 seconds. Thoughts?

Reply:
Maybe you should be able to set a callback to deal with the message? But I think the sensible default here would be to nack the message.

Reply:
I like the callback, smth like: `handle_error(reason, state)`. Then in this callback user has all the power to ack, reject or requeue.

Reply:
btw, are we capturing here also task timeouts?

Reply:
My latest commit adds the ability to configure a timeout for tasks. It was a little tricky to add that in given that Task.Supervisor does not directly support it, but I think I have a good implementation in place to handle this functionality.

As for a custom callback to handle the failure, are you thinking about rewriting the consumer module to a macro so that we can leverage `defoverridable` if the user does not want to use the default error handler?

Reply:
@akoutmos I would prefer to avoid rewriting consumer module to a macro at this stage (just to limit the scope of changes).

Since these changes will be released as a major version, requiring error callback to be implemented should be fine, right? @vorce what do you think?

We could also consider skipping error callback and just reject the messages. The problem here is that users might skip dead-letter configuration for their consumers.

Reply:
Hm hm. Yeah I think making each consumer implement a callback is OK. At least then we encourage users to think about this case. We can have a good implementation in the example consumer that logs and nacks the message.

Reply:
Sounds good. I'll add an additional callback `handle_error` that will be triggered on task exception or task timeout.

Reply:
Awesome, and thanks again for all the work @akoutmos